### DISCUSSION TITLE
Testing with Multiple Transactions

### DISCUSSION THREAD

**Discussion by minaminao** [foundry-rs/foundry#2482] [Help] [answered]
Forge test executes a single transaction, can not we test scenarios with multiple transactions?

For example:
Transaction 1: selfdestruct contract X
Transaction 2: get the code size of contract X (i.e. 0)

Testing with one transaction does not change the code size (i.e. >0).
It seems like it could be done with Anvil, but can it be done with Forge test alone?

(Changing the `block.number` in the cheat code does not change the code size since it is one transaction.)

**Comment by minaminao** [comment, marked as answer]
Since this can be accomplished using Forge Script and it is easy enough to use, I will leave this discussion resolved.

**Reply by gowtham-ponnana** [reply]
How can we do it using forge script? I'm actually looking for solving and searching for hours. If you could link to the exact resource, it would be grateful.

**Reply by Akbar30Bill** [reply]
Well yea. im having kind of the same issue here. we are trying to run a couple of millions of transactions that just does not fit into the forge script transaction gas limit and would like to know how to advance to next transaction or block

**Reply by urosognjenovic** [reply]
How about you bundle all the tests in one function that will execute them all together?

**Reply by Akbar30Bill** [reply]
I actually did this at first then I encountered a gas limit issue since my loop was so big. then, I figured in order to fix this issue I should split the inner loop into multiple functions. But since that wasn't really an option i had to find another way. Then i came across 'vm.pauseGasMetering()' function, which solved my problem altogether. What is function does is it disables all the gas logical altogether.
